### PR TITLE
feat(flutter_tools): Added proxy validator IPV6 loopback check

### PR DESCRIPTION
**Pull request by crisboarna**
Hello, this PR adds IPV6 loopback check in the doctor proxy validator.

*List which issues are fixed by this PR. You must list at least one issue.*
[26587](https://github.com/flutter/flutter/issues/26587)
*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/test/commands.shard/hermetic/proxy_validator_test.dart [hunk 1]
```diff
@@ -37,65 +65,193 @@ void main() {
...
+  @override
+  Uint8List get rawAddress => throw UnimplementedError();
+
+  @override
+  Future<io.InternetAddress> reverse() =>
+    throw UnimplementedError();
+
+  @override
+  io.InternetAddressType get type => throw UnimplementedError();
+}
```
Comment (zanderso) [inline packages/flutter_tools/test/commands.shard/hermetic/proxy_validator_test.dart:257]: nit: need a newline at the end of the file.

**Review by zanderso — APPROVED** [review]
thanks! lgtm w/ nit.

File: packages/flutter_tools/test/commands.shard/hermetic/proxy_validator_test.dart [hunk 2]
```diff
@@ -2,13 +2,41 @@
...
+        ];
+
+        return Future<List<NetworkInterface>>.value(interfaces);
+    });
+  });
+
+  tearDown(() {
+    resetNetworkInterfaceLister();
+  });
+
```
Comment (zanderso) [inline packages/flutter_tools/test/commands.shard/hermetic/proxy_validator_test.dart:39]: The presubmit check is complaining about a trailing whitespace character on this line.

**Comment by zanderso** [comment]
@christopherfujino Do you know why the rebases aren't working for @crisboarna's PR's? The rebase should be moving past the failing rainbowmonkey tests...

**Comment by christopherfujino** [comment]
> @christopherfujino Do you know why the rebases aren't working for @crisboarna's PR's? The rebase should be moving past the failing rainbowmonkey tests...

customer-testing doesn't respect merges. Let me see if I can fix this with a force push...

**Comment by christopherfujino** [comment]
> > @christopherfujino Do you know why the rebases aren't working for @crisboarna's PR's? The rebase should be moving past the failing rainbowmonkey tests...
> 
> customer-testing doesn't respect merges. Let me see if I can fix this with a force push...

Sorry got side-tracked with a p0, let me get back tot he git gymnastics....

**Pull request merged**